client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -1462,6 +1464,7 @@ public void onFailure(Throwable e) {
     try {
       TransportClient client =
           dataClientFactory.createClient(location.getHost(), location.getPushPort(), partitionId);
+      checkClientChanged(mapKey, pushState, client);
```

Comment:
If we need to check for changes every time when create a client, may be using a new method such as createNewClient with parameter (boolean waitPreRequestFinished) is better.

Reply:
yes it's better to create new function that wraps check logic in createClient